src/lib/zork/index.ts

```diff
@@ -94,16 +96,17 @@ interface RTCState {
 
 interface Context {
   clientId: string;
+  legacy: boolean;  // Are we speaking with a legacy uProxy client?
   socket: net.Socket;  // Connection to the zork client. Becomes the signaling channel.
   reply: (msg: string) => void;  // Helper function to reply to the client.
   log: any;  // Helper function for contextual logging.
   mode: Mode;  // null indicates still waiting for a 'give' or 'get' command
-  transformer: any;  // Stores any requested transform settings.
-  socksServer: SocksServer;  // Set if we're getting access.
-  socksSession: SocksSession;  // Set if we're giving access.
-  heartbeatTimeoutId: number;
-  onHeartbeatTimeout: Function;
   rtc: RTCState;
+  // Only used when we're the giver:
+  onHeartbeatTimeout: Function;
+  heartbeatTimeoutId: number;
+  transformer: any;
+  socksSessionByChannelId: {[id: string]: any};
```

Comment:
Proper bookkeeping of socksSession references! since there are several concurrent socks sessions, we need to store them in a collection to be able to retrieve the right one in the channel.onmessage handler.